include/boost/geometry/algorithms/detail/is_simple/linear.hpp

```diff
@@ -54,128 +57,210 @@ namespace detail { namespace is_simple
+        BOOST_ASSERT(boost::size(m_linestring) > 1);
+        return m_is_closed
+            && turn.method == overlay::method_none
+            && check_segment_indices(turn, boost::size(m_linestring) - 2)
+            && geometry::equals(range::front(m_linestring), turn.point);
     }
-};
-
 
+private:
+    Linestring const& m_linestring;
+    bool const m_is_closed;
+};
 
 template <typename MultiLinestring>
-class is_simple_multilinestring
+class is_acceptable_turn<MultiLinestring, multi_linestring_tag>
 {
 private:
-    class is_acceptable_turn
+    typedef typename boost::range_value<MultiLinestring>::type linestring_type;
+    typedef is_acceptable_turn<linestring_type> base_type;
+
+    template <typename Point, typename Linestring>
+    static inline bool is_boundary_point_of(Point const& point,
+                                            Linestring const& linestring)
     {
-    private:
-        template <typename Point, typename Linestring>
-        static inline bool is_boundary_point_of(Point const& point,
-                                                Linestring const& linestring)
-        {
-            BOOST_ASSERT( boost::size(linestring) > 1 );
-            return
-                !geometry::equals(range::front(linestring),
-                                  range::back(linestring))
-                &&
-                ( geometry::equals(point, range::front(linestring))
-                  || geometry::equals(point, range::back(linestring)) );
-        }
+        BOOST_ASSERT(boost::size(linestring) > 1);
+        return
+            ! geometry::equals(range::front(linestring),
+                               range::back(linestring))
+            &&
+            (geometry::equals(point, range::front(linestring))
+             || geometry::equals(point, range::back(linestring)));
+    }
 
-        template <typename Point, typename Linestring>
-        static inline bool is_closing_point_of(Point const& point,
-                                               Linestring const& linestring)
-        {
-            BOOST_ASSERT( boost::size(linestring) > 1 );
-            return
-                geometry::equals(range::front(linestring),
-                                 range::back(linestring))
-                &&
-                geometry::equals(range::front(linestring), point)
-                ;
-        }
+    template <typename Turn, typename Linestring>
+    static inline bool is_closing_point_of(Turn const& turn,
+                                           Linestring const& linestring)
+    {
+        BOOST_ASSERT(boost::size(linestring) > 1);
+        return
+            turn.method == overlay::method_none
+            &&
+            check_segment_indices(turn, boost::size(linestring) - 2)
+            &&
+            geometry::equals(range::front(linestring), range::back(linestring))
+            &&
+            geometry::equals(range::front(linestring), turn.point)
+            ;
+    }
```

Comment:
My point is, you could pass this new policy into the `detail::self_get_turn_points::get_turns<>::apply()`. Here in the implementation of 1-argument `intersects()` https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/intersects.hpp in the lines 57 and 65 the regular (A/A) turn_info and policy types are defined. But you could use the turn_info and policy specific for the pair of input parameters, like it's done in relate https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/relate/turns.hpp in the lines 46 and 41.

Another alternative would be to modify the regular (A/A) policy and just set those position flags if specific turn_info type was used, or always for Linear geometries. The required parameters are passed into all turn policies, in regular policy they're just ignored because Areal geometries doesn't have the front or back (see https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/algorithms/detail/overlay/get_turn_info.hpp line 910).